framework/src/base/Assembly.C

```diff
@@ -2489,7 +2489,11 @@ Assembly::init(const CouplingMatrix * cm)
   if (_block_diagonal_matrix && scalar_vars.size() != 0)
     _block_diagonal_matrix = false;
 
-  auto num_vector_tags = _residual_vector_tags.size();
```

Comment:
This line of code assumes the residual tags are ordered from 0 sequentially, which in general may not be true. The change at the following along the changes in `TimeIntegrator` can fix this temporarily but I believe that somewhere else in the code we have the same assumption. @fdkong can you think of any? I know because if I reset the changes in `TimeIntegrator` I got test failures.

Reply:
We do not expect any order here. It does not need to start from zero.

Reply:
If that caused issue, it should be a bug or something

Reply:
In `Assembly`, we are using `_type_id` for indexing, but in `TaggingInterface`, we use `_id`, which may mismatch `_type_id`. The fix should be in `TaggingInterface`. Just figured that out.